clusterpools/apply.sh

```diff
@@ -510,14 +594,88 @@ if [[ "$CLUSTERPOOL_NAME" == "" ]]; then
     if [[ "$INPUT_CLUSTERPOOL_NAME" == "" ]]; then
         CLUSTERPOOL_NAME="$GENERATED_CLUSTERPOOL_NAME-$(echo $PLATFORM | tr '[:upper:]' '[:lower:]')-$(echo $CLUSTERIMAGESET_NAME | sed 's/\.//g')"
     else
+        CUSTOM_CLUSTERPOOL_NAME="true"
         CLUSTERPOOL_NAME="$INPUT_CLUSTERPOOL_NAME"
     fi
 fi
 printf "${GREEN}* Using Name: $CLUSTERPOOL_NAME${CLEAR}\n"
 
 
+#-----OPTIONALLY CONFIGURE A CUSTOM INSTALL CONFIG SECRET-----#
+if [[ "$YQ_INSTALLED" != "false" ]]; then
+    if [[ "$CUSTOM_CLUSTERPOOL_NAME" == "true" ]]; then
+        CLUSTERPOOL_INSTALL_CONFIG_SECRET_NAME="${CLUSTERPOOL_NAME}-${RANDOM_IDENTIFIER}-install-config-template"
+    else
+        CLUSTERPOOL_INSTALL_CONFIG_SECRET_NAME="${CLUSTERPOOL_NAME}-install-config-template"
+    fi
+    if [[ "$CLUSTERPOOL_INSTALL_CONFIG_FILE" == "" ]]; then
+        printf "${BLUE}- note: to skip this step in the future and create a new install-config secret from a file automatially, export CLUSTERPOOL_INSTALL_CONFIG_FILE${CLEAR}\n"
+        if [[ "$CLUSTERPOOL_INSTALL_CONFIG_SECRET" == "" ]]; then
```

Comment:
Why do you have both `CLUSTERPOOL_INSTALL_CONFIG_SECRET` and `CLUSTERPOOL_INSTALL_CONFIG_SECRET_NAME`?
I'm not immediately seeing a reason to have them be different names...

Reply:
Yepp - so the difference here is that one of them points to the file that the user has on their system that contains the yaml that should be put into a secret on the cluster.  The latter is the name of a secret that they already created that contains the install-config on the cluster that they want us to reference (and skip creating another secret).

Reply:
Aren't you referring to `CLUSTERPOOL_INSTALL_CONFIG_FILE`?

Reply:
Ok, now that I'm sound of mind and not multitasking - what is happening here is:
```
    if [[ "$CLUSTERPOOL_INSTALL_CONFIG_FILE" == "" ]]; then
        printf "${BLUE}- note: to skip this step in the future and create a new install-config secret from a file automatically, export CLUSTERPOOL_INSTALL_CONFIG_FILE${CLEAR}\n"
        if [[ "$CLUSTERPOOL_INSTALL_CONFIG_SECRET" == "" ]]; then
            printf "${BLUE}- note: to skip this step in the future and use a pre-existing install-config secret, export CLUSTERPOOL_INSTALL_CONFIG_SECRET${CLEAR}\n"
```

If the user tells us "here is a file that contains an install config (CLUSTERPOOL_INSTALL_CONFIG_FILE), I would like you to create a secret from it and use that", otherwise if a user says "here is a secret I already have on the cluster (maybe from a previous run) that I would like you to use as the install config", otherwise we prompt them.  How's that sound?

Reply:
I was confused about why we have both `CLUSTERPOOL_INSTALL_CONFIG_SECRET` and `CLUSTERPOOL_INSTALL_CONFIG_SECRET_NAME` and whether if I set the latter if anything would break or if both were really needed.

But `CLUSTERPOOL_INSTALL_CONFIG_SECRET_NAME` is used as a placeholder and would be overwritten here regardless: https://github.com/open-cluster-management/lifeguard/blob/7ffef5b6752115039c246acb6fd7259f466abbb1/clusterpools/apply.sh#L606-L610

And then `CLUSTERPOOL_INSTALL_CONFIG_SECRET` is the official variable used to set the secret name.

I think I get it now. 🙂